### PR TITLE
PY3: Client base classes and utility fixes

### PR DESCRIPTION
**Pull request by klaussfreire**
- Fixes decorators to work correctly in py3 (mostly namespace stuff)
- Modernizes closures to use nonlocal variables instead of the
  "list cell hack"
- Fixes some tests to work reliably in py3

### CODE REVIEW

**Review by gefarion — APPROVED** [review]

File: tests/threadpooltest.py [hunk 1]
```diff
@@ -271,9 +281,19 @@ def testWrapper(self):
         def accounting(i):
             counts[i] += 1
         qsequence = [(mean,"mean"), (simple,"simple")]
+
+        # Ensure full buffering before test start, to make sure
+        # proper batch priorization is taking place, as it won't
```
Comment (chpedersen) [inline tests/threadpooltest.py:286]:
```suggestion
        # proper batch prioritization is taking place, as it won't
```

File: tests/threadpooltest.py [hunk 2]
```diff
@@ -271,9 +281,19 @@ def testWrapper(self):
...
+        # under light load
+        barrier = Event()
+        for i in range(32):
```
Comment (chpedersen) [inline tests/threadpooltest.py:289]: where does this 32 come from?

**Review by chpedersen — APPROVED** [review]

File: tests/threadpooltest.py (hunk 1 above)
Comment (chpedersen) [reply tests/threadpooltest.py:286]: (same below)

File: tests/threadpooltest.py (hunk 2 above)
Comment (klaussfreire) [reply tests/threadpooltest.py:289]: Max number of threads I expect to be running this thread. Totally a guess.

**Pull request merged**